pypeit/wavecalib.py

```diff
@@ -6,44 +6,222 @@
+        for_header = {}
+
+        # Spat_ID first
+        if self.spat_id is None:
+            msgs.error("Cannot write WaveCalib without spat_id")
+        _d.append(dict(spat_id=self.spat_id))
+
+        # Rest of the datamodel
+        for key in self.keys():
+            if key == 'spat_id':
+                continue
+            # Skip None
+            if self[key] is None:
+                continue
+            # Array?
+            if self.datamodel[key]['otype'] == np.ndarray and key != 'wv_fits':
+                _d.append({key: self[key]})
+            elif key == 'wv_fits':
+                for ss, wv_fit in enumerate(self[key]):
+                    # Naming
+                    wv_fit.hdu_prefix = 'SPAT_ID-{}_'.format(self.spat_id[ss])
+                    dkey = 'WAVEFIT-{}'.format(self.spat_id[ss])
+                    # Save
+                    _d.append({dkey: wv_fit})
+            else: # Add to header of the spat_id image
+                _d[0][key] = self[key]
+        #import pdb; pdb.set_trace()
+        # Return
+        return _d
+
+    @classmethod
+    def _parse(cls, hdu, ext=None, transpose_table_arrays=False, debug=False,
+               hdu_prefix=None):
+        # Grab everything but the bspline's
+        _d, dm_version_passed, dm_type_passed = super(WaveCalib, cls)._parse(hdu)
+        # Now the wave_fits
+        list_of_wave_fits = []
+        spat_ids = []
+        for ihdu in hdu:
+            if 'WAVEFIT' in ihdu.name:
+                iwavefit = wv_fitting.WaveFit.from_hdu(ihdu)
+                if iwavefit.version != wv_fitting.WaveFit.version:
+                    msgs.warn("Your WaveFit is out of date!!")
+                list_of_wave_fits.append(iwavefit)
+                # Grab PypeItFit
+                hdname = ihdu.name.replace('WAVEFIT', 'PYPEITFIT')
+                if hdname in [khdu.name for khdu in hdu]:
+                    iwavefit.pypeitfit = fitting.PypeItFit.from_hdu(hdu[hdname])
+                # Grab SPAT_ID for checking
+                i0 = ihdu.name.find('ID-')
+                i1 = ihdu.name.find('_WAV')
+                spat_ids.append(int(ihdu.name[i0+3:i1]))
+        # Check
+        if spat_ids != _d['spat_id'].tolist():
+            msgs.error("Bad parsing of the MasterFlat")
+        # Finish
+        _d['wv_fits'] = np.asarray(list_of_wave_fits)
+        return _d, dm_version_passed, dm_type_passed
+
+    @property
+    def par(self):
+        return json.loads(self.strpar)
+
+
+    def is_synced(self, slits):
+        """
+        Confirm the slits in WaveCalib are aligned to that in SlitTraceSet
+
+        Barfs if not
+
+        Args:
+            slits (:class:`pypeit.slittrace.SlitTraceSet`):
+
+        """
+        if not np.array_equal(self.spat_id, slits.spat_id):
+            msgs.error("Your wvcalib solutions are out of sync with your slits.  Remove Masters and start from scratch")
+
+    def build_waveimg(self, spectrograph, tilts, slits, spat_flexure=None):
```

Comment:
I think this should be a @classmethod. It does not appear to require self anywhere.